wallet/createtx.go

```diff
@@ -1204,6 +1204,11 @@ func (w *Wallet) purchaseTickets(ctx context.Context, op errors.Op,
 		return nil, err
 	}
 
+	// If the ticket price exceeds the max ticket price, skipping purchase
+	if req.MaxTicketPrice > 0 && ticketPrice > req.MaxTicketPrice {
```

Comment:
Given that the new config item is a TicketBuyer config, I don't think this check should be buried deep in the generic `purchaseTickets` func, it should really live in the ticketbuyer code. If the ticket price is too high, ticketbuyer should not be calling purchaseTickets at all. For reference, see how and where the check for balance to maintain is handled.